src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/SyntaxTokenExtensions.cs

```diff
@@ -18,10 +20,17 @@ internal static partial class SyntaxTokenExtensions
         public static bool IsLastTokenOfNode<T>(this SyntaxToken token) where T : SyntaxNode
             => token.IsLastTokenOfNode<T>(out _);
 
-        public static bool IsLastTokenOfNode<T>(this SyntaxToken token, [NotNullWhen(true)] out T node) where T : SyntaxNode
+        public static bool IsLastTokenOfNode<T>(this SyntaxToken token, [NotNullWhen(true)] out T? node) where T : SyntaxNode
```

Comment:
Can we have a unit test? How did you end up finding this issue?

Reply:
from talking to @allisonchou and having her depend on the current behavior in a way that made a PR super confusing for me :)